core/src/input_manager.cpp

```diff
@@ -0,0 +1,207 @@
+#include <GLFW/glfw3.h>
+#include <knoting/input_manager.h>
+#include <knoting/window.h>
+#include <algorithm>
+#include <iostream>
+
+namespace knot {
+
+InputManager::InputManager(Window& owner) : m_sensitivity(0.3f), m_lastScroll(0.0f) {
+    for (int i = 0; i < (int)KeyCode::Last; ++i) {
+        m_keyBinding[i] = false;
+        m_holdKeyBindings[i] = false;
+    }
```

Comment:
if you make the change to std::array this can become m_keyBindings.fill(false);
I would prefer the above!

if not you can do std::fill(m_keyBindings,m_keyBindings + KeyCode::Last, false);

Reply:
done